### ISSUE TITLE
Numpy array record field assignment is broken in numpy 1.9

### ISSUE THREAD

**Issue by MisterSheik** [numpy/numpy#5648]
```python
In [11]: a = np.dtype({'names':['time','data','pad'], 'formats':['<f8',('<f8', (3, 2)),'<f8'], 'offsets':[0,8,56], 'itemsize':64, 'aligned':True})

In [12]: b = np.zeros(10, dtype=a)

In [13]: b[3][1] = np.ones((3, 2))

In [14]: b
Out[14]:
array([(0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[1.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0)],
      dtype={'names':['time','data','pad'], 'formats':['<f8',('<f8', (3, 2)),'<f8'], 'offsets':[0,8,56], 'itemsize':64, 'aligned':True})

In [15]: b[3][1][:] = np.ones((3, 2))
```
But this works:
```
In [16]: b
Out[16]:
array([(0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[1.0, 1.0], [1.0, 1.0], [1.0, 1.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0),
       (0.0, [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0]], 0.0)],
      dtype={'names':['time','data','pad'], 'formats':['<f8',('<f8', (3, 2)),'<f8'], 'offsets':[0,8,56], 'itemsize':64, 'aligned':True})
```

**Comment by charris** [comment]
Duplicate of #2599. There are a bunch of related tickets out there, see also #3126.

**Comment by charris** [comment]
But thanks for the report ;)

**Comment by MisterSheik** [comment]
Thank you Charles for the quick reply.  Please feel free to add my test case to your testing.  It took me a couple hours to track down what stopped working.  It would be nice for this to keep working after (and if) it is fixed :)

**Comment by Hotschke** [comment]
I have checked numpy 1.10.2 and it seems to be fixed in 1.10.2.
